.ado/jobs.yml

```
@@ -22,19 +30,36 @@ jobs:
         ReleaseARM64:
           BuildConfiguration: release
           BuildPlatform: arm64
-
+    variables:
+      semanticVersion: $[ dependencies.Setup.outputs['setVersions.semanticVersion'] ]
+      fileVersion: $[ dependencies.Setup.outputs['setVersions.fileVersion'] ]
     steps:
-      - task: PowerShell@2
-        displayName: Run the build script for publish
-        inputs:
-          targetType: filePath
-          filePath: $(Build.SourcesDirectory)\.ado\scripts\cibuild.ps1
-          arguments:
-            -SourcesPath:$(Build.SourcesDirectory)
-            -OutputPath:$(Build.ArtifactStagingDirectory)
-            -Platform:$(BuildPlatform)
-            -Configuration:$(BuildConfiguration)
-            -AppPlatform:uwp
+      - ${{ if parameters.dependsOnSetup }}:
+        - task: PowerShell@2
+          displayName: Run the build script for publish
+          inputs:
+            targetType: filePath
+            filePath: $(Build.SourcesDirectory)\.ado\scripts\cibuild.ps1
+            arguments:
+              -SourcesPath:$(Build.SourcesDirectory)
+              -OutputPath:$(Build.ArtifactStagingDirectory)
+              -Platform:$(BuildPlatform)
+              -Configuration:$(BuildConfiguration)
+              -AppPlatform:uwp
+              -ReleaseVersion:$(semanticVersion)
+              -FileVersion:$(fileVersion)
+      - ${{ else }}:
+        - task: PowerShell@2
+          displayName: Run the build script for publish
+          inputs:
+            targetType: filePath
+            filePath: $(Build.SourcesDirectory)\.ado\scripts\cibuild.ps1
+            arguments:
+              -SourcesPath:$(Build.SourcesDirectory)
+              -OutputPath:$(Build.ArtifactStagingDirectory)
+              -Platform:$(BuildPlatform)
+              -Configuration:$(BuildConfiguration)
+              -AppPlatform:uwp

```

Sorry GitHub's diff is not very good, is is way better in codeflow...
```suggestion
      - task: PowerShell@2
        displayName: Run the build script for publish
        inputs:
          targetType: filePath
          filePath: $(Build.SourcesDirectory)\.ado\scripts\cibuild.ps1
          arguments:
            -SourcesPath:$(Build.SourcesDirectory)
            -OutputPath:$(Build.ArtifactStagingDirectory)
            -Platform:$(BuildPlatform)
            -Configuration:$(BuildConfiguration)
            -AppPlatform:uwp
            - ${{ if parameters.dependsOnSetup }}:
              -ReleaseVersion:$(semanticVersion)
              -FileVersion:$(fileVersion)
```

---

Okay, github does a horrible job for this suggestion compared to ADO...
Do not apply it, just copy-paste the code....

---

Thanks! I submitted this change from my machine - the change here does not look complete.

---

I had to revert it because when I run it in the publishing pipeline I am getting this error: 
`/.ado/jobs.yml: (Line: 48, Col: 50, Idx: 1508) - (Line: 48, Col: 50, Idx: 1508): Mapping values are not allowed in this context.`

---

I have simplified the code by using double quote for the version parameter values. When the variables are missing - we are getting just empty values and our code builds just fine in such case.